modules/sql/src/main/scala/SqlModule.scala

```diff
@@ -14,8 +14,24 @@ trait SqlModule[F[_]] {
   /** The type of a codec that reads and writes column values of type `A`. */
   type Codec[A]
 
+  /** A codec that has forgotten its type argument. */
+  trait ExistentialCodec {
+    type A
+    def codec: Codec[A]
+  }
+  object ExistentialCodec {
+    def apply[T](c: Codec[T]): ExistentialCodec =
+      new ExistentialCodec {
+        type A = T
+        val codec = c
+      }
+  }
+
+  implicit def unliftCodec(c: ExistentialCodec): Codec[c.A] =
+    c.codec
+
   /** The type of an encoder that writes column values of type `A`. */
-  type Encoder[A]
+  type Encoder[-A]
```

Comment:
If `Encoder` is invariant there are one million pages of type errors in `MappedQuery`. Scala 2 is happy to crush stuff to `Any` after a while but Scala 3 keeps up and doesn't like what's happening in calls to `loop(...)`.